pkg/utils/utils.go

```diff
@@ -286,6 +302,60 @@ func CustomizeEnvSSO(pts *corev1.PodTemplateSpec, instance *openlibertyv1beta1.O
 		if oidcClient.HostNameVerificationEnabled != nil {
 			ssoEnv = append(ssoEnv, *createEnvVarSSO(id, "_HOSTNAMEVERIFICATIONENABLED", *oidcClient.HostNameVerificationEnabled))
 		}
+		// clientId & Secret from registration will replace anything from the *-olapp-sso secret
+		if isOpenShift && oidcClient.AutoRegisterSecret != "" {
+			secretName := oidcClient.AutoRegisterSecret
+			regSecret := &corev1.Secret{}
+			err := client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: instance.GetNamespace()}, regSecret)
+			if err != nil {
+				return errors.Wrapf(err, "Registration Data Secret for Single sign-on (SSO) of name %q was not found. Create a secret in namespace %q with the credentials for the login providers you selected in application image.", secretName, instance.GetNamespace())
+			}
+			theRoute := &routev1.Route{}
+			err = client.Get(context.TODO(), types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}, theRoute)
+			if err != nil {
+				// if route is unavailable, we want to let reconciliation proceed so it will be created,
+				// but we'll update status of the instance so reconcilation will be triggered again.
+				b := false
+				instance.Status.RouteAvailable = &b
+				logf.Log.WithName("utils").Info("CustomizeEnvSSO waiting for route to become available, requeue")
+				return nil
+			}
+		
+			clientId := string(regSecret.Data["RegisteredOidcClientId"])
+			clientSecret := string(regSecret.Data["RegisteredOidcSecret"])
+			
+			if clientId == "" {
+				// if we don't have a client id and secret yet, go get one
+				regData := RegisterData{
+					DiscoveryURL:            oidcClient.DiscoveryEndpoint,
+					RouteURL:                "https://" + theRoute.Spec.Host,
+					RedirectToRPHostAndPort: sso.RedirectToRPHostAndPort,
+				}
+				clientId, clientSecret, err = RegisterWithOidcProvider(regData, regSecret)
+				if err != nil {
+					return errors.Wrapf(err, "Error occured during registration with OIDC provider")
```

Comment:
It'll be good to add the provider ID here.

Reply:
fixed